lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -176,29 +186,43 @@ handleConstraint e = handleJust select handler . fmap Right
 
 ----------------------------------------------------------------------------
 -- Database layer methods
+newDBLayer
+    :: forall s t. (W.IsOurs s, NFData s, Show s, PersistState s, W.TxId t)
+    => Maybe FilePath
+       -- ^ Database file location, or Nothing for in-memory database
+    -> IO (DBLayer IO s t)
+newDBLayer = fmap snd . newDBLayer'
+
 
 -- | Sets up a connection to the SQLite database.
 --
 -- Database migrations are run to create tables if necessary.
 --
 -- If the given file path does not exist, it will be created by the sqlite
 -- library.
-newDBLayer
+newDBLayer'
     :: forall s t. (W.IsOurs s, NFData s, Show s, PersistState s, W.TxId t)
     => Maybe FilePath
        -- ^ Database file location, or Nothing for in-memory database
-    -> IO (DBLayer IO s t)
-newDBLayer fp = do
+    -> IO (Sqlite.Connection, DBLayer IO s t)
+newDBLayer' fp = do
     lock <- newMVar ()
     bigLock <- newMVar ()
-    conn <- createSqliteBackend fp (dbLogs [LevelError])
+    (conn, backend) <- createSqliteBackend fp (dbLogs [LevelError])
     let runQuery' :: SqlPersistM a -> IO a
-        runQuery' = withMVar bigLock . const . runQuery conn
+        runQuery' cmd = case fp of
+            Nothing ->
+                withMVar bigLock $ const $ runQuery backend cmd
+            _ ->
+                withMVar bigLock $ const $ runResourceT $ runNoLoggingT $ withSqlConn (createSqliteBackend1 fp)
```

Comment:
file-based, this cannot be used with in-memory. Needs other pairs of eyes - at this moment do not see why